src/udev/net/ethtool-util.c

```diff
@@ -612,9 +612,11 @@ int ethtool_set_glinksettings(int *fd, const char *ifname, struct link_config *l
         if (link->port != _NET_DEV_PORT_INVALID)
                 u->base.port = link->port;
 
-        u->base.autoneg = link->autonegotiation;
+        if (link->autonegotiation != -1)
```

Comment:
btw, instead of comparing explicitly with `!= -1` here, let's just filter out negative values, i.e. i.e. change `!= -1` to `>= 0`